signal_analog/flow.py

```diff
@@ -142,6 +146,85 @@ def find_publish(statement):
             raise ProgramDoesNotPublishTimeseriesError(statements)
 
 
+class Plot(object):
+    """Represents a Plot for a Chart as configured in the SignalFx UI.  It is a helper class that makes it simpler
+       to use lower-level abstractions like Assign, Function, Data, publish()"""
+
+    def __init__(self, assigned_name, signal_name, filter=None, rollup=None, fx=None, label=None, visible=True,
+                 extrapolation_policy=None, max_extrapolations=None):
+        """Represents a Plot for a Chart as configured in the SignalFx UI.  It is a helper class that makes it simpler
+        to use lower-level abstractions like Assign, Function, Data, publish()
+
+        Other options available in the UI for a plot include: display_units, plot_color, and visualization_type.  These
+        values can be configured in with_publish_label_options() in the Chart itself.  See PublishLabelOptions.
+
+        Example:
+
+        >>> TimeSeriesChart() \
+        >>>    .with_name("Cpu Utilization") \
+        >>>    .with_program(
+        >>>       Plot("A", "CPUUtilization", filter, rollup="max", fx=[Mean(by="app")])
+        >>> )
+
+        Arguments:
+            assigned_name: the assigned name used in the SignalFlow program, use A-Z to keep compatibility with the
+                    builder in the SignalFx UI. E.g. the "A" in "A = data('metric_name')"
+            signal_name: the metric name, e.g. "CPUUtilization"
+            filter: the filter to apply to the metric, e.g. And(Filter("env, "prod"), Filter("app", "foo"))
+            rollup: String If None then the default rollup for the metric is used. Otherwise one of the following
```

Comment:
Might be easier to use an `enum` here since the kinds of rollup values are fixed. It'll prevent annoying errors too where you put `avg` instead of `average` and scratch your head wondering why it isn't working 🤔

Doesn't have to be anything fancy either, it can be as simple as this:
https://github.com/Nike-Inc/signal_analog/blob/749915745cccb5fbef75212027bde8040a44e0ea/signal_analog/charts.py#L155-L160

I'm undecided if this documentation should move to the enum or not, but I'll leave that up to you. Does this sound like a reasonable change?

Reply:
Enum is better.  I was just following the API that was used elsewhere.  I can change it though.

Reply:
I've added a RollupType enumeration.  I think I've done it in a way that is backward compatible (most people are already passing in strings).